src/Bowling.js

```diff
@@ -0,0 +1,144 @@
+
+"use strict";
+
+const ALL_PINS = 10;
+
+function Bowling() {
+  this.score = 0;
+  this.frameKey;
+  this.frameScore;
+  this.spare = false;
+  this.strike = false;
+  this.input 
+  //this.rollCount = 1   maybe this should be instead of passing rollCount;
+}
+
+function getRndInteger(min, max) {
+  return Math.floor(Math.random() * (max - min)) + min;
+}
+
+Bowling.prototype.runCardMaking = function (input) {
+  var rollCount;
+  var frameCount;
+  var pins;
+  var score;
+  this.makeCardTemplate();
+  for (frameCount = 1; frameCount < 11; frameCount++) {
+    this.frameKey = frameCount;
+    for (rollCount = 1; rollCount < 3; rollCount++) {
+      if (this.isSkipped(rollCount)) {
+        pins = "x";
+        score = this.calculateScore(0);
+      } else {
+       
+        pins = this.input;
+
+        this.switchStrike(rollCount, pins)
+         
+        this.scoreStrike(rollCount, pins);
+        score = this.calculateScore(pins);
+      }
+      this.fillCard(pins, score, rollCount);
+
+      if (this.strike === false && this.spare === true) {
+        this.scoreSpares(rollCount, pins);
+      }
+      this.switchSpare();
+    }
+  }
+  // end of loop big loop
+  console.log(this.card)
+  return this.card;
+};
+
+Bowling.prototype.isSkipped = function (rollCount) {
+  if (rollCount === 2 && this.strike === true) {
+    return true;
+  }
+};
+
+Bowling.prototype.makeCardTemplate = function () {
+  /// changed obj but perhaps was better before as means I have to make conditions for runCardmaking
+  var frameCount,
+    key,
```

Comment:
Would separating methods linked to individual frames into a frame object help here?